### PR TITLE
Mapping info

### PR DESCRIPTION
**Pull request by bergbauer**
@peterrum

### CODE REVIEW

File: include/deal.II/non_matching/mapping_info.h [hunk 1]
```diff
@@ -64,20 +69,70 @@ namespace NonMatching
...
+      const IteratorRange<Iterator> &             cell_iterator_range,
+      const std::vector<std::vector<Point<dim>>> &unit_points_vector,
+      const unsigned int n_unfiltered_cells = numbers::invalid_unsigned_int);
+
+    template <typename Iterator>
+    void
+    reinit_cells(
+      const IteratorRange<Iterator> &     cell_iterator_range,
+      const std::vector<Quadrature<dim>> &quadrature_vector,
+      const unsigned int n_unfiltered_cells = numbers::invalid_unsigned_int);
```
Comment (peterrum) [inline include/deal.II/non_matching/mapping_info.h:92]: ?

File: include/deal.II/non_matching/mapping_info.h [hunk 2]
```diff
@@ -157,30 +261,548 @@ namespace NonMatching
...
+                 "This mapping info is not reinitialized for a cell vector"));
+        if (do_cell_index_compression)
+          {
+            Assert(
+              cell_index_to_compressed_cell_index[cell_index] !=
+                numbers::invalid_unsigned_int,
+              ExcMessage(
+                "Mapping info object was not initialized for this active cell "
+                "index"));
+            auto it_begin = unit_points.begin() +
```
Comment (peterrum) [inline include/deal.II/non_matching/mapping_info.h:683]:
```suggestion
            const auto it_begin = unit_points.begin() +
```

File: include/deal.II/matrix_free/fe_point_evaluation.h [hunk 3]
```diff
@@ -470,6 +470,20 @@ class FEPointEvaluation
...
+   */
+  void
+  reinit(const unsigned int cell_index);
+
+  /**
+   * Reinitialize the evaluator to point to the correct precomputed mapping of
+   * the face in the MappingInfo object.
+   */
+  void
+  reinit(const unsigned int cell_index, const unsigned int face_number);
```
Comment (peterrum) [inline include/deal.II/matrix_free/fe_point_evaluation.h:485]: Can we keep the name general? `cell_index` and `face_index` is quite specific. These are rather indices into the `MappingInfo`.

**Review by peterrum — COMMENTED** [review]
Looks very nice :+1:

**Review by peterrum — COMMENTED** [review]
Looks very good. Feel free to a PR in deal.II!

File: include/deal.II/matrix_free/fe_point_evaluation.h (hunk 3 above)
Comment (bergbauer) [reply include/deal.II/matrix_free/fe_point_evaluation.h:485]: What would you suggest?

**Comment by bergbauer** [comment]
Let's continue in the deal.II PR!

**Pull request closed without merging**